### ISSUE TITLE
Syncthing: After installation and configuration -> 503 Service Unavailable

### ISSUE THREAD

**Issue by kalakrumanov** [freedombox/Plinth#940]
Plinth is now throwing this on ALL the apps (I can access to Plinth normally, and the configuration, but neither Transmission, Syncthing or any other app loads)

![screen shot 2017-08-01 at 9 53 57 pm](https://user-images.githubusercontent.com/16352883/28844263-570c24be-7704-11e7-8df1-f388c0467601.png)

I'm using raspberry pi 1, with image freedombox-testing-nonfree_0.9_raspberry-armel.

When I type: `sudo service plinth status`, I get this:

```
plinth.service - Plinth Web Interface
   Loaded: loaded (/lib/systemd/system/plinth.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2017-07-12 16:01:50 CEST; 2 weeks 6 days ago
     Docs: man:plinth(1)
 Main PID: 443 (plinth)
   CGroup: /system.slice/plinth.service
           └─443 /usr/bin/python3 /usr/bin/plinth
```
And in the logs, everything is looking fine, no errors. 

How do I stop the "maintenance downtime"? Or, what is happening?

**Comment by JosephKiranBabu** [comment]
Do you see any errors in the apache error log at /var/log/apache2/error.log or in the Plinth status log at /var/log/plinth/status.log when this happens?

**Comment by kalakrumanov** [comment]
Here is the apache error log when I try to go to /transmission

> [Wed Aug 02 09:53:48.726548 2017] [proxy:error] [pid 14671] (111)Connection refused: AH00957: HTTP: attempt to connect to [::1]:9091 (localhost) failed
> [Wed Aug 02 09:53:48.727050 2017] [proxy_http:error] [pid 14671] [client 192.168.1.45:56814] AH01114: HTTP: failed to make connection to backend: localhost, referer: https://192.168.1.67/plinth/apps/transmission/

And this is the status plinth as you requested:

> [2017-08-02 09:35:16,368] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/transmission', 'get-configuration']

**Comment by SunilMohanAdapa** [comment]
Looking at the apache log, it looks like the transmission daemon did not accept the connection from Apache.  This is likely to happen if transmission daemon is not running.

Does transmission page show that the daemon is running?  What about the diagnostic results? `systemctl status transmission-daemon`?

We had situations where Raspberry Pi 1 ran out of its 512 MiB RAM when some we tried to use certain services. Could it have been that memory ran out on the system and kernel killed a bunch of services leading to multiple services not being available?  Would be nice to know what you were doing when this happened. In this case restarting the system should bring back the services. Plinth will get restarted always after a kill (due to systemd configuration) but may not happen for other application.

**Comment by kalakrumanov** [comment]
This is what I get in the Apache error log:

_Trying to access localhost/syncthing_

> [Wed Aug 02 15:08:57.402556 2017] [proxy:error] [pid 17895] (111)Connection refused: AH00957: HTTP: attempt to connect to [::1]:8384 (localhost) failed
> [Wed Aug 02 15:08:57.403043 2017] [proxy_http:error] [pid 17895] [client 192.168.1.45:60083] AH01114: HTTP: failed to make connection to backend: localhost

_Trying to access localhost/transmission_

> [Wed Aug 02 15:09:16.339315 2017] [proxy:error] [pid 14675] (111)Connection refused: AH00957: HTTP: attempt to connect to [::1]:9091 (localhost) failed
> [Wed Aug 02 15:09:16.339675 2017] [proxy_http:error] [pid 14675] [client 192.168.1.45:60086] AH01114: HTTP: failed to make connection to backend: localhost

And as you asked, this is what I get when typing `systemctl status transmission-daemon`

> ● transmission-daemon.service - Transmission BitTorrent Daemon
>    Loaded: loaded (/lib/systemd/system/transmission-daemon.service; enabled; vendor preset: enabled)
>    Active: active (running) since Wed 2017-07-12 16:01:52 CEST; 2 weeks 6 days ago
>  Main PID: 435 (transmission-da)
>    Status: "Idle."
>    CGroup: /system.slice/transmission-daemon.service
>            └─435 /usr/bin/transmission-daemon -f --log-error
> 
> Jul 12 16:01:50 freedombox systemd[1]: Starting Transmission BitTorrent Daemon...
> Jul 12 16:01:52 freedombox systemd[1]: Started Transmission BitTorrent Daemon.
> Jul 12 16:01:53 freedombox transmission-daemon[435]: [2017-07-12 16:01:53.167] UDP Failed to set receive buffer: requested 4194304, got 327680 (tr-udp.c:84)
> Jul 12 16:01:53 freedombox transmission-daemon[435]: [2017-07-12 16:01:53.167] UDP Failed to set send buffer: requested 1048576, got 327680 (tr-udp.c:95)
>

**Comment by kalakrumanov** [comment]
It looks like my Rpi is getting out of RAM. This is what I get when typing `free -m` :

>                      total       used        free  shared  buff/cache  available
> Mem:            434         164          12          19         257         241
> Swap:             0           0           0

**Comment by kalakrumanov** [comment]
IT WORKS NOW.

I have made some tweakings in Mysql conf, rebooted the RP1 and now I have more free RAM than before, and everything is working fine again.

FYI, I made the following changes in `/etc/mysql/my.cnf` (just added the following lines at the end of the file)

> key_buffer              = 8M
> max_connections         = 30
> query_cache_size        = 8M
> query_cache_limit       = 512K
> thread_stack            = 128K
> performance_schema = ON
> innodb_log_file_size should be = 16M
> innodb_buffer_pool_instances =1

Thanks for your answers and patience ;)

**Comment by SunilMohanAdapa** [comment]
@kalakrumanov Did you spot which process was taking up most of the RAM.  We can perhaps warn our users.  Usually, all the default services don't occupy a lot of RAM and the typical usage would be about 100 MiB.

**Comment by kalakrumanov** [comment]
Yes, MYSQLD is eating a lot of RAM. Take a look at this screenshot:

![screen shot 2017-08-09 at 4 43 08 pm](https://user-images.githubusercontent.com/16352883/29127643-f3b72796-7d21-11e7-9095-042ed318d88e.png)

There must be a way for it not using soooo much RAM. What do you think @SunilMohanAdapa ?

**Comment by JosephKiranBabu** [comment]
Maybe if we can find out which Plinth application is using Mysql, maybe we can consider replacing it with sqlite3. 
One more thing to take care of is to have only one DB, either Postgres or Mysql for Plinth applications.

**Comment by kalakrumanov** [comment]
I have installed some selfhosted projects aside form Plinth, but none of them are used by NOBODY except me. 

That could be the problem @JosephKiranBabu  ?

**Comment by kalakrumanov** [comment]
And "phpmyadmin" also, as "Adminer" gave me some errors.

**Comment by JosephKiranBabu** [comment]
I'm not aware of any Plinth application using MySql. Even for Diaspora, I've configured it to use Postgresql since other applications like tt-rss are using it too. We should probably stick to Postgresql even for future applications, i.e. only if Sqlite3 doesn't satisfy the need or isn't supported by that application.

**Comment by kalakrumanov** [comment]
Sorry, I haven't said that I installed it later, MariaDB, for "playing around" with some other selfhosted projects, such as "PrivateBin" or "Laverna" (great open source projects, btw).

So is "my fault" that it takes so much ram.
